tests/helpers/test_check_config.py

```diff
@@ -246,3 +247,23 @@ async def test_config_platform_raise(hass: HomeAssistant) -> None:
         assert err.domain == "bla"
         assert err.message == "Unexpected error calling config validator: Broken"
         assert err.config == {"value": 1}
+
+
+async def test_removed_yaml_support(hass: HomeAssistant) -> None:
+    """Test config validation check with removed CONFIG_SCHEMA without raise if present."""
+    mock_integration(
+        hass,
+        MockModule(
+            domain="bla", config_schema=cv.removed("bla", raise_if_present=False)
+        ),
+        False,
+    )
+    files = {YAML_CONFIG_FILE: BASE_CONFIG + "bla:\n  platform: demo"}
+    with patch("os.path.isfile", return_value=True), patch_yaml_files(files):
+        res = await async_check_ha_config_file(hass)
+        log_ha_config(res)
+
+        assert res.keys() == {"homeassistant"}
+        assert res.error_str.startswith(
+            "Config found for bla integration in configuration.yaml. YAML support for this integration has been removed. Please remove it from your configuration.yaml."
+        )
```

Comment:
I guess this test will now fail?

Reply:
Oh, yes. I'm on mobile now so I can fix this when I'm home